lib/heap_sort.rb

```diff
@@ -1,8 +1,30 @@
-
+require_relative "max_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n log n)
+# Space Complexity: O(log n) - since using recurisve heap down
 def heap_sort(list)
```

Comment:
This works but since you're building a heap with n nodes, the space complexity is O(n).